napari/_vispy/layers/image.py

```diff
@@ -156,6 +159,10 @@ def _on_interpolation_change(self):
             else self.layer.interpolation3d
         )
 
+    def _on_custom_interpolation_kernel_change(self):
+        if self.layer._slice_input.ndisplay == 2:
+            self.node.custom_kernel = self.layer.custom_interpolation_kernel
```

Comment:
From where it is called when change ndisplay from 3d to 2d? Is reset function called in such situation?

Reply:
yes, at the end of `_on_display_change`, reset is called.